src/dodal/beamlines/i04.py

```diff
@@ -359,10 +369,56 @@ def oav_to_redis_forwarder() -> OAVToRedisForwarder:
 @device_factory()
 def diamond_filter() -> DiamondFilter[I04Filters]:
     """Get the i04 diamond filter device, instantiate it if it hasn't already been.
-    If this is called when already instantiated in i03, it will return the existing object.
+    If this is called when already instantiated in i04, it will return the existing object.
     """
     return DiamondFilter[I04Filters](
         prefix=f"{PREFIX.beamline_prefix}-MO-FLTR-01:",
         name="diamond_filter",
         data_type=I04Filters,
     )
+
+
+@device_factory()
+def panda_fast_grid_scan() -> PandAFastGridScan:
+    """Get the i04 panda_fast_grid_scan device, instantiate it if it hasn't already been.
+    If this is called when already instantiated in i04, it will return the existing object.
+    This is used instead of the zebra_fast_grid_scan device when using the PandA.
+    """
+    return PandAFastGridScan(
+        prefix=f"{PREFIX.beamline_prefix}-MO-SGON-01:",
+        name="panda_fast_grid_scan",
+    )
+
+
+@device_factory()
+def panda() -> HDFPanda:
+    """Get the i04 panda device, instantiate it if it hasn't already been.
+    If this is called when already instantiated in i04, it will return the existing object.
+    """
+    return HDFPanda(
+        f"{PREFIX.beamline_prefix}-EA-PANDA-01:",
+        path_provider=get_path_provider(),
+        name="panda",
+    )
+
+
+@device_factory()
+def zocalo() -> ZocaloResults:
+    """Get the i04 ZocaloResults device, instantiate it if it hasn't already been.
+    If this is called when already instantiated in i04, it will return the existing object.
+    """
+    return ZocaloResults(
+        name="zocalo",
+        prefix=PREFIX.beamline_prefix,
+    )
```

Comment:
Should: `ZocaloResults` doesn't need a prefix and as above, name will get filled in for us, so we could just do:
```suggestion
    return ZocaloResults()
```

Reply:
Have done this